code/weapon/beam.cpp

```diff
@@ -1037,6 +1039,7 @@ void beam_move_all_post()
 	beam *next_one;	
 	int bf_status;	
 	beam_weapon_info *bwi;
+	int type_d_index, type_d_wait = 0;
```

Comment:
Variables should be defined as close as possible to the usage.

Reply:
I was under a mistaken impression that there were reasons to avoid declaring variables inside a loop. That impression has been dispelled, I'll move them